pennylane/qnode.py

```diff
@@ -750,11 +750,12 @@ def tape(self) -> QuantumTape:
     def construct(self, args, kwargs):  # pylint: disable=too-many-branches

```

Comment:
```suggestion
    def construct(self, args, kwargs, shots=None):  # pylint: disable=too-many-branches
```
Potentially it might be good to keep `shots` separate from "things in the qfunc's call signature".  But I guess we can go either way.

Reply:
I didn't do this purely to avoid changing the function interface. If it's not a big deal, I would actually prefer the interface you recommend. I don't know if anyone has strong opinions about the change.

Reply:
I think this is due to the subtlety of a user who might define a qfunc with `shots` as an argument. Hence the `kwargs.pop("shots")` below

Reply:
And then we can write out the call signature for `construct` like:


```
Args:
    args (tuple): positional arguments for the user-provided quantum function
    kwargs (dict): keyword arguments for the user-provided quantum function
    shots (None, int, tuple[int], tuple[int, tuple[int]]): the shots to use for the execution. 

Sets:
     ``tape``: the circuit to be executed
     ``_qfunc_output``: the output of the quantum function.

Raises:
    ``QuantumFunctionError``
```